galaxy_ng/app/migrations/_dab_rbac.py

```diff
@@ -39,18 +39,28 @@ def split_pulp_roles(apps, schema_editor):
     GroupRole = apps.get_model('core', 'GroupRole')
 
     for corerole in Role.objects.all():
+        print(f'ROLE {corerole} {corerole.name}')
         split_roles = {}
         for assignment_cls in (UserRole, GroupRole):
+            print(f'\t{assignment_cls}')
             for pulp_assignment in assignment_cls.objects.filter(role=corerole, content_type__isnull=False):
+                print(f'\t\t{assignment_cls} {pulp_assignment}')
                 if pulp_assignment.content_type_id not in split_roles:
+                    print(f'\t\t\t{pulp_assignment.content_type_id}')
                     new_data = {
                         'description': corerole.description,
                         'name': f'{corerole.name}_{pulp_assignment.content_type.model}'
                     }
                     new_role = Role(**new_data)
                     new_role.save()
+
+                    # add the permission back? ...
+                    for perm in pulp_assignment.role.permissions.all():
+                        new_role.permissions.add(perm)
```

Comment:
Yes, this was my oversight and this is the shape of the change I would have made. The only change I'd like is to skip any `perm` that has `perm.codename` not valid for the particular content type. That would be:

```python
from ansible_base.rbac.validators import permissions_allowed_for_role, combine_values

cls = pulp_assignment.content_type.model_class()
ct_codenames = combine_values(permissions_allowed_for_role(cls))
if perm.codename not in ct_codenames:
  continue
```

With test_app, example:

```
In [5]: combine_values(permissions_allowed_for_role(Namespace))
Out[5]: 
{'add_collectionimport',
 'change_collectionimport',
 'change_namespace',
 'delete_collectionimport',
 'delete_namespace',
 'view_collectionimport',
 'view_namespace'}
```